### PR TITLE
Plane: support hygrometer data for DroneCAN airspeed sensors

### PR DESCRIPTION
**Pull request by tridge**
This supports DroneCAN airspeed sensors which can provide hygrometer data for supporting de-icing status.
The CUAV skye airspeed sensor can provide this data
https://www.cuav.net/en/_skye/
I originally planned to do this driver as a lua script, but ended up incorporating it into the airspeed driver on 2M flash boards, as the space needed is not overly large

### CODE REVIEW

**Comment by IamPete1** [comment]
We should be able to improve our air density estimate using humidity, can make a big difference in warm air. I don't think baro's do this already?

**Comment by tridge** [comment]
> We should be able to improve our air density estimate using humidity, can make a big difference in warm air. I don't think baro's do this already?

hmm, might be worth doing?
@priseborough is this a big enough effect for speed scaling ?

**Comment by IamPete1** [comment]
It can be quite a big effect, the hotter the air the bigger difference. 

image


From: https://en.wikipedia.org/wiki/Density_of_air

**Comment by tridge** [comment]
> It can be quite a big effect, the hotter the air the bigger difference

so at max temperature we could really fly at (45C or so?) it might be 5%?

**Comment by IamPete1** [comment]
3.5% reduction in density between 0% and 100% humidity at 45 deg c at seal level. So not a vast change, but if we have a reliable reading we might as well apply the correction.

File: libraries/AP_Airspeed/AP_Airspeed_UAVCAN.h [hunk 1]
```diff
@@ -26,20 +27,33 @@ class AP_Airspeed_UAVCAN : public AP_Airspeed_Backend {
...
     static void handle_airspeed(AP_UAVCAN* ap_uavcan, uint8_t node_id, const AirspeedCb &cb);
+    static void handle_hygrometer(AP_UAVCAN* ap_uavcan, uint8_t node_id, const HygrometerCb &cb);
 
     static AP_Airspeed_UAVCAN* get_uavcan_backend(AP_UAVCAN* ap_uavcan, uint8_t node_id);
 
     float _pressure; // Pascal
     float _temperature; // Celcius
     uint32_t _last_sample_time_ms;
 
+    // hygrometer data
```
Comment (tridge) [inline libraries/AP_Airspeed/AP_Airspeed_UAVCAN.h:50]: add ifdef on data

**Review by peterbarker — APPROVED** [review]

**Comment by rmackay9** [comment]
This is included in Copter-4.3.1-rc1

**Pull request merged**